### PR TITLE
feat(Manager) : Integrating Lighthouse | revamping web properties and spa properties page

### PR DESCRIPTION
**Pull request by nikhitamore**
## Explain the feature/fix

1. Lighthouse integration
2. Web properties page layout changed
3. spa detailed page revamped to new designs

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/manager/src/services/lighthouse/queries.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  list: (webPropertyIdentifier: string, applicationIdentifier: string, env: string = '') =>
+    ['lighthouse', webPropertyIdentifier, applicationIdentifier, env] as const
+};
+
+const fetchLighthouseReportForGivenBuildId = async (
+  propertyIdentifier: string,
+  identifier: string,
+  env: string,
+  buildId: string
+): Promise<any> => {
```
Comment (soumyadip007) [inline packages/manager/src/services/lighthouse/queries.tsx:16]: @nikhitamore  Add a todo comment to change the any to interface later

File: packages/manager/src/services/lighthouse/queries.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    {
+      params: {
+        skip: 'lhBuildId'
+      }
+    }
+  );
+
+  console.log('fetchLighthouseReportForGivenBuildId', data.data[0].lhProjectId, buildId, env);
+  // lighthouse_dev_static_2
+
```
Comment (soumyadip007) [inline packages/manager/src/services/lighthouse/queries.tsx:28]: remove these lines

File: packages/manager/src/services/lighthouse/queries.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  try {
+    const { data } = await orchestratorReq.get(`/applications/property/${propertyIdentifier}`, {
+      params: {
+        applicationIdentifier,
+        env
+      }
+    });
+    const pipelineDetails = data.data && data.data.length > 0 ? data.data[0].pipelineDetails : [];
+    // TODO: To be removed after backend revamp
+    // pipeline array will be empty for new spa, for existing sopa if no report is generated then key xpipelineDetails won't be there
```
Comment (soumyadip007) [inline packages/manager/src/services/lighthouse/queries.tsx:56]: change `xpipelineDetails` to `pipelineDetails`

File: packages/manager/src/services/lighthouse/queries.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  useQuery(lighthouseKeys.list(webPropertyIdentifier, applicationIdentifier, env), () =>
+    fetchLhIdentifierList(webPropertyIdentifier, applicationIdentifier, env)
+  );
+
+// POST Operations
+export const generateLighthouseReport = async (
+  dto: TLighthouseGenerateDTO
+): Promise<TCreateApiKeyRes> => {
+  const { data } = await orchestratorReq.post('/lighthouse/generate', dto);
+  console.log('data', data);
```
Comment (soumyadip007) [inline packages/manager/src/services/lighthouse/queries.tsx]: remove the log

File: packages/manager/src/views/SPAPropertyDetailPage/Lighthouse/Lighthouse.tsx [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+  const getColor = (percentage: number): string => {
+    if (percentage >= 90) {
+      return '#6EC664'; // Green
+    } else if (percentage >= 50) {
+      return '#EF9234'; // Orange
+    }
+    return '#C9190B'; // Red
+  };
+
+  // const generateLighthouseReport = useGenerateLighthouseReport();
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/Lighthouse/Lighthouse.tsx:103]: remove these line

File: packages/manager/src/views/SPAPropertyDetailPage/Lighthouse/Lighthouse.tsx [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+  const metricNames = ['performance', 'accessibility', 'bestPractices', 'seo', 'pwa'];
+
+  const getColor = (percentage: number): string => {
+    if (percentage >= 90) {
+      return '#6EC664'; // Green
+    } else if (percentage >= 50) {
+      return '#EF9234'; // Orange
+    }
+    return '#C9190B'; // Red
+  };
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/Lighthouse/Lighthouse.tsx:101]: TODO : add the color code to CSS file later

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment.tsx/StaticSPADeployment.tsx [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+  Spinner,
+  Title,
+  Tooltip
+} from '@patternfly/react-core';
+import { Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { AxiosError } from 'axios';
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import toast from 'react-hot-toast';
+// import './StaticSPADeployment.css';
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment.tsx/StaticSPADeployment.tsx:37]: remove the unused import

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment.tsx/StaticSPADeployment.tsx [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+  const onClose = () => {
+    setIsExpanded(false);
+  };
+
+  const onSelectDataListItem = (id: string) => {
+    const index = parseInt(id.charAt(id.length - 1), 10);
+    const rowSelectedData = paginatedData && paginatedData[index];
+    setSelectedData(rowSelectedData);
+    setSelectedDataListItemId(id);
+    // setDrawerPanelBodyContent(index + 1);
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment.tsx/StaticSPADeployment.tsx:133]: remove unused code

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment.tsx/StaticSPADeployment.tsx [hunk 9]
```diff
@@ -0,0 +1,355 @@
...
+    setIsExpanded(true);
+  };
+
+  //   const paginatedData = staticDeploymentData?.slice(startIdx, endIdx);
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment.tsx/StaticSPADeployment.tsx:137]: remove this comment

**Review by soumyadip007 — CHANGES_REQUESTED** [review]
@nikhitamore please make these changes

File: packages/manager/src/services/lighthouse/queries.tsx (hunk 2 above)
Comment (nikhitamore) [reply packages/manager/src/services/lighthouse/queries.tsx]: done

File: packages/manager/src/services/lighthouse/queries.tsx (hunk 1 above)
Comment (nikhitamore) [reply packages/manager/src/services/lighthouse/queries.tsx:18]: done

File: packages/manager/src/services/lighthouse/queries.tsx (hunk 3 above)
Comment (nikhitamore) [reply packages/manager/src/services/lighthouse/queries.tsx:55]: done

**Review by soumyadip007 — APPROVED** [review]

**Pull request merged**